src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -384,6 +384,45 @@ public abstract class SendReceive<T> : SocketTestHelperBase<T> where T : SocketH
             }
         }
 
+        [OuterLoop]
+        [Theory]
+        [MemberData(nameof(Loopbacks))]
+        public void SendRecv_BlockingStreamRead_Close(IPAddress listenAt)
+        {
+            // Test for #22564 when close is called while blocking read is pending.
+            using (var server = new Socket(listenAt.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+            using (var client = new Socket(listenAt.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+            {
+                server.BindToAnonymousPort(listenAt);
+                server.Listen(1);
+
+                Thread clientThread = new Thread(() =>
+                {
+                    byte[] buffer = new byte[16];
+                    client.Connect(server.LocalEndPoint);
+                    // Blocking read.
+                    try
+                    {
+                        client.Receive(buffer);
+                    }
+                    catch (ObjectDisposedException)
+                    {
+                        // Closed and disposed by parent thread.
+                        return;
+                    }
+                    catch (SocketException) { };
+
+                    client.Close();
+                });
+                clientThread.IsBackground = true;
+                clientThread.Start();
+                Socket s = server.Accept();
+                // Close client socket from parent thread.
+                client.Close();
```

Comment:
Why is the client.Close duplicated here as well?

Reply:
This is way how to force close from other thread. The clientThread should be blocked in receive() . This mimics example from #26034